file.go

```diff
@@ -1,37 +1,121 @@
 package jcsv
 
-import "os"
+import (
+	"encoding/csv"
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"os"
+)
 
 type file struct {
 	// TODO: define variables if needed
+	JSONFormat map[string]interface{}
+	CSVFormat  struct {
+		Data [][]string
+		//if "headers==true", data[0] will contain headers
+		HasHeaders bool
+	}
 }
 
+func (f *file) initialize() {
+	f.JSONFormat = nil
+	f.CSVFormat.Data = nil
+	f.CSVFormat.HasHeaders = false
+}
 func ParseJsonFile(path string) (file, error) {
 	// TODO: open and read the given file into your `file` object
-	return file{}, nil
+	jsonFile, err := os.Open(path)
+	var myFile file
+	myFile.initialize()
+	if err != nil {
+		return myFile, err
+	}
+	defer jsonFile.Close()
+	byteValue, err := ioutil.ReadAll(jsonFile)
+	if err == nil {
+		json.Unmarshal([]byte(byteValue), &myFile.JSONFormat)
```

Comment:
`ioutil.ReadAll()` already returns `[]byte`. What is the need for typecasting this?

Reply:
It is redundant, I will resolve this.